### ISSUE TITLE
[HOLD for payment 2023-05-02] [$4000]  mweb/safari - Request money - viewport is jumping up and down when navigating menus

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#15270]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Access staging.new.expensify.com
2. Sign into a valid account
3. Tap on FAB > Request Money > Navigate the currency & menus

## Expected Result:
User expects that the viewport and buttons remain in a fixed position

## Actual Result:
The buttons & viewport are jumping and moving around when opening and closing the keyboard

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.74.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/219829185-e4c5cf6a-9b44-4c3e-b2bb-d76af8187e5f.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~015abb72f6bcd9af5a</li>
        <li>Upwork Job ID: 1630330487569076224</li>
        <li>Last Price Increase: 2023-03-13</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @strepanier03 (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by strepanier03** [comment]
Was OoO yesterday for President's day, will handle today.

**Comment by strepanier03** [comment]
@puneetlath - Would you mind taking a look at this GH and letting me know if your GH [here](https://github.com/Expensify/App/issues/14716) is related? After reading through your GH and the solution and behaviors for iOS m/web described [here](https://github.com/Expensify/App/issues/14716#issuecomment-1419964610) seem similar. 

I'm torn between adding external because it's easily reproducible behavior and waiting to see if the PR for your GH fixes this or alters the behavior.

**Comment by puneetlath** [comment]
Hm, it could be yeah. Maybe let's put this on hold until that PR hits staging and then we can see if it solved the problem or not.

**Comment by strepanier03** [comment]
Thank you @puneetlath I will do that and follow along with the PR progress.

**Comment by strepanier03** [comment]
Retesting since the other PR is in prod now.

**Comment by strepanier03** [comment]
All right, this is still occurring so sending on and removing hold status.

**Comment by strepanier03** [comment]
Testing:

Able to recreate with the testing steps in the OG comment.

https://user-images.githubusercontent.com/10925636/221696919-bfaa7b37-83b5-47e6-8335-8f5d15b88ee2.mp4

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~015abb72f6bcd9af5a

**Comment by MelvinBot** [comment]
Current assignee @strepanier03 is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @thesahindia (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @hayata-suenaga (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by PankajAS** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
View of Amount page height shrink and grow when navigate from currency selection

### What is the root cause of that problem?
Root cause of that problem is `flex:1` which is filling available space in IOS/safari after navigate because parent view don't have `minHeight`.

### What changes do you think we should make in order to solve the problem?
We have to add height or `minHeight` for `IOUAmountPage` component's parent view.

https://user-images.githubusercontent.com/17126628/221826421-051384f9-dd3b-4ef5-bd75-9aa10523450a.mp4


### What alternative solutions did you explore? (Optional)
None

**Comment by dennismunene** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The buttons & viewport are jumping and moving around when opening and closing the keyboard

### What is the root cause of that problem?
Setting the max height  for the `KeyboardAvoidingView` via the "styles" prop may cause the view to disappear or flicker on certain devices.

### What changes do you think we should make in order to solve the problem?
We need to use "contentContainerStyle" prop instead of the "style" prop for the KeyboardAvoidingView.

Edit this line
https://github.com/Expensify/App/blob/c0f14f444a62fe8d0f70425f36251d2f962496ba/src/components/ScreenWrapper/index.js#L121
 to this

```
<KeyboardAvoidingView style={[styles.w100, styles.h100]}
                               contentContainerStyle={{ maxHeight:  this.props.windowHeight }} 
                               behavior={this.props.keyboardAvoidingViewBehavior}>
```


https://user-images.githubusercontent.com/4717301/222435057-90943e94-77be-4ebd-84a3-f6a1a8a3dbf1.mp4

**Comment by PankajAS** [comment]
> This isn't an iou amount page specific issue. If the keyboard is open while navigating to a page that has a button at bottom then the same behaviour will be reproducible there. I added a text input at initial settings page and you can notice that now we have the same behaviour at workspace list page

yes @thesahindia as i mention in [Proposal](https://github.com/Expensify/App/issues/15270#issuecomment-1447928018) in safari(ios) where we set `flex:1` it will jump in safari to fill space and to fix this we need to add `minHeight`

**Comment by dennismunene** [comment]
@thesahindia  With my current proposal ,The button does not stay above the keyboard when the keyboard is open and a user is selected at the new group page. 
What is the expected behaviour here ?

**Comment by hayata-suenaga** [comment]
@thesahindia yes I agree if a past issue has solved the problem of hidden button, we don't want to hide the button.

**Comment by strepanier03** [comment]
Head's up that I'm out of the office tomorrow so if there is an urgent need for BugZero action on this GH, please reach out in #expensify-open-source to ask for help. 

I'll be back Monday and will check in then.

**Comment by MelvinBot** [comment]
@strepanier03 @thesahindia @hayata-suenaga this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by hayata-suenaga** [comment]
Still reviewing proposals

**Comment by hayata-suenaga** [comment]
I assume both @PankajAS's [proposal](https://github.com/Expensify/App/issues/15270#issuecomment-1447928018) and @dennismunene's [proposal](https://github.com/Expensify/App/issues/15270#issuecomment-1451828573) are going to hide the button under the keyboard on certain pages.

Should we figure out which pages can the button be hidden under the keyboard and only implement one of their solutions to these pages?

**Comment by dennismunene** [comment]
@hayata-suenaga @thesahindia  I've updated my proposal with a suggested work around that avoids hiding the button on some pages. Let me hear your thoughts on this.

**Comment by dennismunene** [comment]
> > My suggestion is that we wrap pages that use a Form with a parent 'View' and use it to set the maxHeight of the page like this.
> 
> @dennismunene, I think you wanted to say **"pages that don't use `Form` component"** ?
> 
> I tried the solution it will cause a regression. The loading indicator is not in the center.
> 
> @daraksha-dk, We wanna keep the button above the keyboard on some pages e.g. new group and split bill.

@thesahindia Adding `flex:1` to the parent 'View' here should fix the regression on the loading indicator.

**Comment by hayata-suenaga** [comment]
@dennismunene I'm looking at your solution, but I believe `contentContainerStyle` is not used except on iOS native app. [Here](https://github.com/Expensify/App/blob/3c57501f3791e43fb6a1476e4a40ff9ff2f5fbd2/src/components/KeyboardAvoidingView/index.js#L9), you can find that `contentContainerStyle` is omitted from props before they're passed to `KeyboardAvoidingView`.

I don't why your solution worked though when you tested locally... could you double check?

**Comment by dennismunene** [comment]
@hayata-suenaga I doubled checked and you were right. `contentContainerStyle` is omitted and never used.Thank you for pointing that out. 
Removed that line from my proposed solution and re-tested locally, seems to be working fine.

**Comment by hayata-suenaga** [comment]
@dennismunene thank you. i'll check your solution on my end tomorrow

**Comment by MelvinBot** [comment]
@strepanier03 @thesahindia @hayata-suenaga this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a> and will need to go internal. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by strepanier03** [comment]
Proposal discussion still ongoing. @hayata-suenaga @thesahindia - Do you think we're getting close to a proposal we can accept?

**Comment by hayata-suenaga** [comment]
@strepanier03 we're still seeking solutions that don't cause a regression and that don't require adding certain lines to new pages (i.e. wrapping components and setting `maxHeight`).

I'll create a new thread on slack to dump this and get more proposals.

**Comment by hayata-suenaga** [comment]
Created a thread [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1678730956613939).

**Comment by MelvinBot** [comment]
📣 @gedu You have been assigned to this job by @mountiny!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by mountiny** [comment]
Eduardo from Callstack will help with this.

@hayata-suenaga @thesahindia Eduardo will still post a proposal how to fix this before they make a PR.

**Comment by hayata-suenaga** [comment]
I'll wait for @gedu's proposal

**Comment by strepanier03** [comment]
Thanks @mountiny - I'll keep following along.

**Comment by gedu** [comment]
Looking into a better way trying to avoiding wrapping and passing an extra prop

**Comment by gedu** [comment]
@hayata-suenaga 

## Proposal

**Please re-state the problem that we are trying to solve in this issue.**
The issue we need to address is related to the Number Pad jumping on mobile web when there is a change in the viewport height.

**What is the root cause of that problem?**
The root cause of this issue is that the IOUModal root component is ScreenWrapper, which is designed to listen to changes in Dimensions. When the height changes, a new maxHeight value is set. Although this is the expected behavior in most cases, it is causing a problem for the request money feature. As a result, we need to lock the height to prevent the components from moving.

**What changes do you think we should make in order to solve the problem?**
To fix this issue, I propose adding a new prop to the ScreenWrapper component. This new prop, named “lockHeight,” will provide us with more control over the maxHeight setting. By default, it will be set to false, which means that it will be updated on every dimension change. However, we can set it to true only if the device accepts touch, and the current step is IOUAmount, to ensure the request money screen flow works as expected.

**What alternative solutions did you explore?**
I attempted to address the issue by using ResizeObserver to listen to height changes and improve component placement since KeyboardAvoidView does not work on mobile browsers. However, my investigation with ResizeObserver was inconclusive despite indications on the website that it should be supported. Although further investigation is needed to make it work, I chose to use the current solution to avoid delaying the fix. Even if ResizeObserver did work, we would need a new prop to enable or disable the observation.

**Comment by MelvinBot** [comment]
@gedu @strepanier03 @thesahindia @hayata-suenaga this issue is now 4 weeks old and preventing us from maintaining WAQ, can you:
- Decide whether any proposals currently meet our guidelines and can be approved as-is today
- If no proposals meet that standard, please take this issue internal and treat it as one of your highest priorities
- If you have any questions, don't hesitate to start a discussion in #expensify-open-source

Thanks!

**Comment by MelvinBot** [comment]
Current assignee @thesahindia is eligible for the Internal assigner, not assigning anyone new.

**Comment by hayata-suenaga** [comment]
@gedu thank you for the proposal. could you elaborate on "lock the height to prevent the components from moving"? Are we going to pass `maxHeight` to `ScreenWrapper` when setting `lockHeight` to true?

And for other screens where this jumping behavior is observed, should we also set `lockHeight` to true and pass `maxHeight` to `ScreenWrappers`?

Which screens / components should we set the new props?

**Comment by gedu** [comment]
@hayata-suenaga 

> could you elaborate on "lock the height to prevent the components from moving"?
Yes, because `ScreenWrapper` is composed of `withWindowDimensions`, and in this case, all the screens are shown in the same `IOUModal` when the keyboard shows up the `maxHeight` style changes. And because the `Dimensions.addEventListener` [notifies the height changes](https://github.com/Expensify/react-native-web/pull/10) when the keyboard animation is over, you see the component jump. So, with this new variable, `lockHeight` my intention is to use the max height set by Dimension when the component renders, preventing when the keyboard shows up to set a smaller height.

> And for other screens where this jumping behavior is observed, should we also set lockHeight to true and pass maxHeight to ScreenWrappers?

Yes, for other screens when the component jump because of the height change, setting the `lockHeight` to true should be enough. The `maxHeight` is given by `withWindowDimensions` so that value always will be observed and passed down, I was thinking to ignore the new height given by `withWindowDimensions`.

> Which screens / components should we set the new props?

Not sure, which screens exactly, I can do a quick passthrough

**Comment by hayata-suenaga** [comment]
@gedu thank you for your answers 🙏 I'd appreciate it if you could clarify one more point for me.

> lockHeight my intention is to use the max height set by Dimension when the component renders the first time, preventing when the keyboard shows up to set a smaller height.

So with your proposal, screens that change height after the keyboard dismissal animation maintain the height that accounts for keyboard?

For example, will the amount page has this UI?

<img width="318" alt="Screenshot 2023-03-21 at 9 32 22 AM" src="https://user-images.githubusercontent.com/98560306/226678536-7ee9a791-4e5d-40ca-aebc-e34c282dbdc4.png">

instead of this?

<img width="312" alt="Screenshot 2023-03-21 at 9 32 33 AM" src="https://user-images.githubusercontent.com/98560306/226678669-f80bf82f-15ef-4ff3-965e-d4a13f0c90bd.png">

**Comment by gedu** [comment]
@thesahindia 

> I am not sure if the solution matches with our expected behaviour.

The fix I'm proposing will behave like this


https://user-images.githubusercontent.com/1676818/226888005-18df4630-f424-4f82-8516-3c2b59432c11.mp4

Will work for the cases you commented

**Comment by hayata-suenaga** [comment]
as long as the video goes, it looks like it's following the expected behavior. @thesahindia can you also confirm this? if it's okay, @gedu can raise a PR for this.

**Comment by situchan** [comment]
I think the solution should be more general. This behavior happens in many places.
I don't think removing `maxHeight` is the correct and general solution. It was added intentionally in #14392. When we have input on both pages, it might cause regression of #10670 or #11463.
I recommend to fix this in navigation level.

i.e. FAB icon position after closing Search page:

https://user-images.githubusercontent.com/108292595/227787064-af0b1022-7753-4d64-8dba-f1282833bf18.mp4

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by hayata-suenaga** [comment]
@situchan thank you for the continued investigation. @gedu is working on the PR for this. @gedu could you check @situchan's comment?

**Comment by hayata-suenaga** [comment]
@gedu's PR addresses several screens, but maybe not the debit card addition page, too. @gedu could you also check that you're fix does not cause regression that @thesahindia mentioned above?

**Comment by gedu** [comment]
@hayata-suenaga @thesahindia 

This jumping issue will reappear on several screens. because many screens use the `ScreenWrapper`. After this [intentional fix](https://github.com/Expensify/App/pull/14392/files).
It seems to be a misconception regarding where to place the fix and what it does. I believe it should be in the "KeyboardAvoidingView", ideally, it should only wrap components that need to avoid the keyboard. However, in the previous fix, the whole screen was set to "avoid" the keyboard always, which only changed the height and resulted in unexpected behavior.
To adhere to the philosophy of "KeyboardAvoidingView", I propose a solution where we use 100% of the height and ignore the height limitation. While there may be other ways to improve this and create an overall solution, ultimately, it will come down to selecting which components need to be moved and which ones don't, so it will be a page-by-page case.

Here is placing the fix to the Payment:

https://user-images.githubusercontent.com/1676818/229150027-bc48354e-3c52-48d5-82ac-dd94a4d2548c.mp4


While I'm trying my best to learn everything as quickly as possible, there may be certain concepts or screens that I am not yet familiar with. In such cases, I may need some guidance or clarity on how you would like to handle them.

**Comment by gedu** [comment]
@thesahindia About this #15543, I can apply my fix, so the button will be at the bottom all the time, and I can check the height so I only allow the button to move if the device is not `isSmallScreenWidth`

**Comment by hayata-suenaga** [comment]
@thesahindia if you could create a list of screens that are mentioned throughout this GH issue comments for needing a fix for @gedu, I would really appreciate it so that we don't miss any.

**Comment by gedu** [comment]
@thesahindia Super, thanks for taking the time to get the info. Will work on them.

About the `Froms`, do you have or is already a config value that I can use to check the "blank" area between the Keyboard and the form is big enough for the button, that I can use?

**Comment by gedu** [comment]
@thesahindia @hayata-suenaga About handling the button when is in a Form, I will add an `onLayout` to the Form so I can get the height of the children, and then use that to know if I should enable the maxHeight or not.

**Comment by gedu** [comment]
@thesahindia @hayata-suenaga I was thinking something like this:


https://user-images.githubusercontent.com/1676818/230592040-6cc4b779-d03e-438b-8270-f8f7d20b4bd4.mp4

I just remove the avatar to make some space, so the `ScreenWrapper` (with help of `Form`) "detects" that has space and enables the maxHeight so the button moves. Now for the math, I'm checking that the content should be less than 40% of the screen if it means there is space for the button. but we can tweak it.

**Comment by hayata-suenaga** [comment]
@gedu thank for the suggestion. @thesahindia can you take a look at the above suggestion?

but yes we can handle this in a separate PR or issue.

**Comment by cubuspl42** [comment]
Hey guys,

I'm trying to propose a solution for [a bug](https://github.com/Expensify/App/issues/16082) introduced by [the fix](https://github.com/Expensify/App/pull/15100) caused by [the bug](https://github.com/Expensify/App/issues/14716) introduced by [the fix](https://github.com/Expensify/App/pull/14392) to [a problem](https://github.com/Expensify/App/issues/11463).

I can see that you're working on the bottom of that chain here.

To solve my bug (the flickering), what would help me would be isolating the original `maxHeight` solution to the "New room" screen. That would, hopefully, leave the app without a regression, while you fight the bigger battle here.

Would you be alright with that?

**Comment by hayata-suenaga** [comment]
@gedu could you check @cubuspl42's comment above?

**Comment by gedu** [comment]
@cubuspl42 Hey, I read the comments a couple of times, and your final summary. I agree with most of your findings, I got confused about your fix, in your formal proposal you said, to wait to merge all possible fixes and later remove `blink_input_opacity_to_prevent_scrolling_when_focus`. but in the [commit](https://github.com/cubuspl42/App/commit/9139c749fc0c9d5f10680e8b87c24f4724f60ee1) there is code that you don't explain and you didn't remove the animation, and in your final sketch didn't get what is your new proposal (if any). I think would be better to restate the proposal explaining with more detail your approach, which is on top of this issue/PR.

cc: @hayata-suenaga

**Comment by cubuspl42** [comment]
> @cubuspl42 Hey, I read the comments a couple of times, and your final summary. I agree with most of your findings, I got confused about your fix,

I'm sorry about the confusion, but I'm not surprised about it. This whole thing is very confusing and I might've accidentally created a bit more confusion.

> in your formal proposal you said, to wait to merge all possible fixes and later remove `blink_input_opacity_to_prevent_scrolling_when_focus`.

Yes, that's correct. The whole proposal in its **current** state can be summed up as "Let's remove these few lines, and here is why that's a good idea...".

> but in the [commit](https://github.com/cubuspl42/App/commit/9139c749fc0c9d5f10680e8b87c24f4724f60ee1) there is code that you don't explain and you didn't remove the animation,

That commit is not part of any PR or proposal, it was an illustration for a discussion. The commit predates the decision to wait for your PR (https://github.com/Expensify/App/pull/16443). Now it's not needed.

> and in your final sketch didn't get what is your new proposal (if any). I think would be better to restate the proposal explaining with more detail your approach, which is on top of this issue/PR.

The whole sketching thing is an attempt to ensure I'm on the same page with @s77rt. [He says](https://github.com/Expensify/App/issues/16082#issuecomment-1502610109):

> But ultimately I'm looking for another fix to https://github.com/Expensify/App/issues/11463

While in the beginning I also was a bit skeptical to the `maxHeight` approach, after understanding all the messy details of Safari's iOS behavior I don't think that anything can be done there, without hacks. I'm also not sure what are the actual properties of a proposal we're potentially waiting for. I'm interested in that because as the external contributor, I'd prefer to improve my proposal so it's acceptable for being selected 🙂

**Comment by strepanier03** [comment]
@gedu @hayata-suenaga @cubuspl42 - Any update here or next steps we can take to move things forward?

**Comment by hayata-suenaga** [comment]
@gedu's PR is almost ready. @gedu could you check again @cubuspl42's comments that the change he is suggesting and your PR doesn't conflict?

**Comment by strepanier03** [comment]
@gedu - How are we looking on this one? Any update that can be provided?

**Comment by gedu** [comment]
@strepanier03  Merged

**Comment by strepanier03** [comment]
Splendid, thank you @gedu!

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.4-0 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/16443


If no regressions arise, payment will be issued on 2023-05-02. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@thesahindia] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@thesahindia] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@thesahindia] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@strepanier03] Determine if we should create a regression test for this bug.
- [ ] [@thesahindia] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@strepanier03] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by strepanier03** [comment]
@dylanexpensify is going to take over this GH for me since it will need action before I return from OoO and I don't want this to be held up on my return.

Thank you Dylan 🙌

**Comment by dylanexpensify** [comment]
Missed this! Let's pay some people!

**Comment by dylanexpensify** [comment]
@thesahindia mind giving me an overview of this one and tackling the checklist items?

**Comment by dylanexpensify** [comment]
@thesahindia @gedu please apply for the job [here](https://www.upwork.com/jobs/~01a4b5c5d8cabd9eb2)! 🙇

**Comment by TomaszFrackowiak** [comment]
@dylanexpensify - hey, gedu is from Callstack :-)

**Comment by dylanexpensify** [comment]
@thesahindia sent offer!

**Comment by dylanexpensify** [comment]
@thesahindia payment sent, contract ended
RT [here](https://github.com/Expensify/Expensify/issues/281905)

**Comment by dylanexpensify** [comment]
@thesahindia can you please complete the checklist items assigned to you? 🙇‍♂️
